beacon-chain/blockchain/process_block_helpers.go

```diff
@@ -129,6 +128,94 @@ func (s *Service) sendLightClientFeeds(cfg *postBlockProcessConfig) {
 	}
 }
 
+// saveLightClientUpdate saves the light client update for this block
+// if it's better than the already saved one, when feature flag is enabled.
+func (s *Service) saveLightClientUpdate(cfg *postBlockProcessConfig) {
+	//if !features.Get().EnableLightClient {
+	//	return
+	//}
+
+	attestedRoot := cfg.roblock.Block().ParentRoot()
+	attestedBlock, err := s.getBlock(cfg.ctx, attestedRoot)
+	if err != nil {
+		log.WithError(err).Error("Could not get attested block")
+		return
+	}
+	if attestedBlock == nil {
+		log.Error("Attested block is nil")
+		return
+	}
+	attestedState, err := s.cfg.StateGen.StateByRoot(cfg.ctx, attestedRoot)
+	if err != nil || attestedState == nil {
```

Comment:
```suggestion
	if err != nil || attestedState == nil || attestedState.IsNil() {
```
It might be better yet to separate errors into two parts, just like you did for the attested block.